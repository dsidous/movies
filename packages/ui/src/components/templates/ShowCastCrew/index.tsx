import React from 'react';

import Spinner from '../../atoms/Spinner';
import FullCastCrew from '../../molecules/FullCastCrew';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';
import { Show } from '@typesRoots/show';

interface Props {
  loading: boolean;
  show: Show;
}

const CastCrew: React.FC<Props> = ({ show, loading }) => {
  if (loading) {
    return <Spinner />;
  }

  const { id, title, name, release_date, first_air_date, poster_path } = show;
  const media = name ? 'tv' : 'movie';

  return (
    <PageTransition>
      <MiniHeader
        title={title || name}
        release_date={release_date || first_air_date}
        poster_path={poster_path}
        link={`/${media}/${id}`}
        linkCopy='Back to main'
      />
      <FullCastCrew movie={show} />
    </PageTransition>
  );
};

export default CastCrew;
