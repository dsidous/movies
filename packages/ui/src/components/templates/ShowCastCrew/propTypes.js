import { object, bool } from 'prop-types';

export const contextTypes = {
  router: object.isRequired,
};

export const propTypes = {
  show: object,
  loading: bool,
};

export const defaultProps = {};
