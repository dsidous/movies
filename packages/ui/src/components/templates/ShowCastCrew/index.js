/* eslint-disable camelcase */
import React from 'react';

import { propTypes } from './propTypes';
import Spinner from '../../atoms/Spinner';
import FullCastCrew from '../../molecules/FullCastCrew';
import PageTransition from '../../atoms/PageTransition';
import MiniHeader from '../../molecules/MiniHeader';

const CastCrew = ({ show, loading }) => {
  if (loading) {
    return <Spinner />;
  }

  const { id, title, name, release_date, first_air_date, poster_path } = show;
  const media = name ? 'tv' : 'movie';

  return (
    <PageTransition>
      <MiniHeader
        title={title || name}
        release_date={release_date || first_air_date}
        poster_path={poster_path}
        link={`/${media}/${id}`}
        linkCopy="Back to main"
      />
      <FullCastCrew movie={show} />
    </PageTransition>
  );
};

CastCrew.propTypes = propTypes;

export default CastCrew;
