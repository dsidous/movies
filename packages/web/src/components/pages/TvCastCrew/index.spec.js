import React from 'react';
import { shallow } from 'enzyme';
import ShowCastCrew from '.';

const defaultProps = {
  match: { params: { tvId: 123 } },
};

describe('Pages/TvCastCrew', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ShowCastCrew {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
