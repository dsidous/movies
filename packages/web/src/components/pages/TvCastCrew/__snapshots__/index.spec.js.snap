// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/TvCastCrew should render as expected 1`] = `ShallowWrapper {}`;
